util/options_test.cc

```diff
@@ -701,7 +710,54 @@ TEST_F(OptionsTest, GetPlainTableOptionsFromString) {
   ASSERT_NOK(GetPlainTableOptionsFromString(table_opt,
              "user_key_len=66;bloom_bits_per_key=20;hash_table_ratio=0.5;"
              "encoding_type=kPrefixXX",
-             &new_opt));  
+             &new_opt));
+}
+#endif  // !ROCKSDB_LITE
+
+#ifndef ROCKSDB_LITE  // GetMemTableRepFactoryFromString is not supported
+TEST_F(OptionsTest, GetMemTableRepFactoryFromString) {
+  MemTableRepFactory* new_mem_factory = nullptr;
+
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "skip_list", &new_mem_factory));
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "skip_list:16", &new_mem_factory));
+  ASSERT_EQ(std::string(new_mem_factory->Name()), "SkipListFactory");
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "skip_list:16:invalid_opt", &new_mem_factory));
+
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "prefix_hash", &new_mem_factory));
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "prefix_hash:1000", &new_mem_factory));
+  ASSERT_EQ(std::string(new_mem_factory->Name()), "HashSkipListRepFactory");
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "prefix_hash:1000:invalid_opt", &new_mem_factory));
+
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "hash_linkedlist", &new_mem_factory));
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "hash_linkedlist:1000", &new_mem_factory));
+  ASSERT_EQ(std::string(new_mem_factory->Name()), "HashLinkListRepFactory");
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "hash_linkedlist:1000:invalid_opt", &new_mem_factory));
+
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "vector", &new_mem_factory));
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "vector:1024", &new_mem_factory));
+  ASSERT_EQ(std::string(new_mem_factory->Name()), "VectorRepFactory");
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "vector:1024:invalid_opt", &new_mem_factory));
+
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "cuckoo", &new_mem_factory));
+  ASSERT_OK(GetMemTableRepFactoryFromString(
+      "cuckoo:1024", &new_mem_factory));
+  ASSERT_EQ(std::string(new_mem_factory->Name()), "HashCuckooRepFactory");
+
+  ASSERT_NOK(GetMemTableRepFactoryFromString(
+      "invalid_factory", &new_mem_factory));
```

Comment:
I think this test will fail under valgrind since the pointers the are created in GetMemTableRepFactoryFromString are not deleted